File: Week2 Arrays/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

// prototype
int letters(string text);
int words(string text);
int sentences(string text);


//main
int main(void)
{
    string text = get_string("Text: ");
    int l = letters(text);
    int s = sentences(text);
    int w = words(text);

    float L = ((float)l / (float)w) * 100;
    float S = ((float)s / (float)w) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index * 1);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}


// define function
// function_letters
int letters(string text)
{
    int letter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            letter++;
        }
    }
    return letter;
}

// function_sentences
int sentences(string text)
{
    int sentence = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence++;
        }
    }
    return sentence;
}

// function_words
int words(string text)
{
    int word = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            word++;
        }
    }
    return word + 1;
}

File: Week6 Python/readability.py
def main():
    text = input('Text: ')
    letter = count_letters(text) * 100 / count_words(text)
    sentence = count_sentences(text) * 100 / count_words(text)
    grade = round(0.0588 * letter - 0.296 * sentence - 15.8)

    if grade < 1:
        print('Before Grade 1')
    elif grade > 16:
        print('Grade 16+')
    else:
        print(f'Grade {grade}')


def count_letters(text):
    cnt = 0
    for i in range(len(text)):
        if text[i].isalnum():
            cnt += 1
    return cnt


def count_sentences(text):
    cnt = 0
    for i in range(len(text)):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            cnt += 1
    return cnt


def count_words(text):
    cnt = 0
    for i in range(len(text)):
        if text[i].isspace():
            cnt += 1
    return cnt + 1


if __name__ == '__main__':
    main()